### PR TITLE
ddl: add new cancel test framework and rewrite some tests

### LINKED ISSUE #33930: enhance cancel test
## Enhancement
There are many tests related to admin cancel ddl, it's better to put these tests together and use a general framework to test them.

### LINKED ISSUE #33933: admin cancel success but the DDL success too.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
Fnn3AA3cPh

If we cancel before running the job on the writeReorg state, we can see that the DDL can't be canceled.

### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: close #33930 and close #33933

Problem Summary:

### What is changed and how it works?
1. Rewrite some cancel tests to cancel_test.
2. Check isReorgCanceled after receiving from doneCh so that the worker can know it's canceled.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/db_partition_test.go [hunk 1]
```diff
@@ -2779,167 +2776,6 @@ LOOP:
...
-				break
-			}
-			step := 10
-			rand.Seed(time.Now().Unix())
-			// delete some rows, and add some data
-			for i := count; i < count+step; i++ {
-				n := rand.Intn(count)
-				tk.MustExec("delete from t1 where c1 = ?", n)
-				tk.MustExec("insert into t1 values (?, ?, ?)", i+10, i, i)
-			}
```
Comment (tangenta) [inline ddl/db_partition_test.go:2865]: The concurrent data deletion is removed?

Comment (wjhuang2016) [reply ddl/db_partition_test.go:2865]: Yes, I don't think we need to modify data for the cancellation.

File: ddl/reorg.go [hunk 2]
```diff
@@ -241,6 +241,14 @@ func (w *worker) runReorgJob(t *meta.Meta, reorgInfo *reorgInfo, tblInfo *model.
 	// wait reorganization job done or timeout
 	select {
 	case err := <-w.reorgCtx.doneCh:
+		if err != nil {
+			w.reorgCtx.clean()
+			return errors.Trace(err)
+		}
+		if w.reorgCtx.isReorgCanceled() {
```
Comment (xiongjiwei) [inline ddl/reorg.go:248]: why need check reorg canceled here? it will return `dbterror.ErrCancelledDDLJob` if the job is cancled

Comment (wjhuang2016) [reply ddl/reorg.go:248]: It may return ok even if reorgCancel is set.

**Review by xiongjiwei — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**